src/group/mls_group/apply_commit.rs

```diff
@@ -145,11 +145,7 @@ impl MlsGroup {
         key_schedule.add_context(&provisional_group_context)?;
         let provisional_epoch_secrets = key_schedule.epoch_secrets(true)?;
 
-        let mls_plaintext_commit_auth_data =
-            match MlsPlaintextCommitAuthData::try_from(mls_plaintext) {
-                Ok(mpcad) => mpcad,
-                Err(_) => return Err(ApplyCommitError::ConfirmationTagMissing),
-            };
+        let mls_plaintext_commit_auth_data = MlsPlaintextCommitAuthData::from(mls_plaintext);
```

Comment:
I think this might have been why I kept it a `try_from`. We probably still want to fail here if the tag is missing in the `mls_plaintext` even if we can now just put the `None` in the `MlsPlaintextCommitAuthData` as well.

Reply:
Good catch! We should definitely fail when there's no ConfirmationTag. I'll revert it to `try_from` and add a failure condition for when no tag is present. Also, a test, of course.